### PR TITLE
Add logistic neighbours

### PR DESCRIPTION
**Pull request by duncan-clark**
added both the custom baselevel to the logistic statistic as wells as adding the logisticNeighbor statistic with the same functionality.

### CODE REVIEW

File: inst/include/Stats.h [hunk 1]
```diff
@@ -1076,8 +1076,8 @@ class Logistic : public BaseStat< Engine > {
 protected:
 	int nstats; /*!< the number of stats generated */
 	int nlevels;
-	int variableIndex, regIndex;
-	std::string variableName, regressorName;
+	int variableIndex, regIndex,baseIndex;
```
Comment (ifellows) [inline inst/include/Stats.h:1079]: I thought I already saw these changes in a pull request?

File: tests/testthat/test-stats.R [hunk 2]
```diff
@@ -34,7 +35,168 @@ test_that("Stats", {
...
+    logistic_test_5 <- (r_stat_5 == cpp_stat_5)
+    logistic_test_6 <- (r_stat_6 == cpp_stat_6)
+    logistic_test_7 <- (r_stat_7 == cpp_stat_7)
+    
+    # ================
+    # Logistic Neighbors
+    # ================
+    
+    # Calculate for base level = "2"
+    r_stat_1 = sum(as.numeric(net %v% "var_2_neighbors")[which(net %v% "var_2" == "1")])
```
Comment (ifellows) [inline tests/testthat/test-stats.R:113]: "<-" not "="

File: tests/testthat/test-stats.R [hunk 3]
```diff
@@ -34,7 +35,168 @@ test_that("Stats", {
...
+    cpp_stat_1 = ernm::calculateStatistics(net ~ logisticNeighbors("var_2","var_2","2") | var_2)
+    
+    # Calculate for base level = "1"
+    r_stat_2 = sum(as.numeric(net %v% "var_2_neighbors")[which(net %v% "var_2" == "2")])
+    cpp_stat_2 = ernm::calculateStatistics(net ~ logisticNeighbors("var_2","var_2","1") | var_2)
+    
+    # Test over all the nets to be sure:
+    r_stat_3 = T
```
Comment (ifellows) [inline tests/testthat/test-stats.R:121]: use TRUE not T

File: tests/testthat/test-stats.R [hunk 4]
```diff
@@ -34,7 +35,168 @@ test_that("Stats", {
...
+        model$setNetwork(as.BinaryNet(net))
+        model$calculate()
+        
+        stat1 <- model$statistics()
+        model$dyadUpdate(2,i)
+        stat2 <- model$statistics()
+        change_2 <- stat2-stat1
+        
+        if(change_1 != change_2){
+            r_stat_4 = F
```
Comment (ifellows) [inline tests/testthat/test-stats.R:160]: FALSE not F

File: tests/testthat/test-stats.R [hunk 5]
```diff
@@ -34,7 +35,168 @@ test_that("Stats", {
...
+        if(change_1 != change_2){
+            r_stat_5 = F
+        }
+    }
+    
+    logistic_neighbot_test_1 <- (r_stat_1 == cpp_stat_1)
+    logistic_neighbot_test_2 <- (r_stat_2 == cpp_stat_2)
+    logistic_neighbot_test_3 <- (r_stat_3 == cpp_stat_3)
+    logistic_neighbot_test_4 <- (r_stat_4 == cpp_stat_4)
+    logistic_neighbot_test_5 <- (r_stat_5 == cpp_stat_5)
```
Comment (ifellows) [inline tests/testthat/test-stats.R:198]: I don't see any actual comparison checks here. Are some expect_*() calls missing?

Comment (ifellows) [reply tests/testthat/test-stats.R:198]: for both logistic and logisticNeighbors

File: inst/include/Stats.h [hunk 6]
```diff
@@ -1177,7 +1224,214 @@ class Logistic : public BaseStat< Engine > {
...
+    void dyadUpdate(const BinaryNet<Engine>& net, int from, int to){
+        bool addingEdge = !net.hasEdge(from,to);
+        //get the variables
+        int fromVarValue = net.discreteVariableValue(variableIndex,from)-1;
+        int fromRegValue = net.discreteVariableValue(regIndex,from)-1;
+        int toVarValue = net.discreteVariableValue(variableIndex,to)-1;
+        int toRegValue = net.discreteVariableValue(regIndex,to)-1;
+        // If we are removing the edge may remove a value for both from  and to
+        if(addingEdge){
+            if(fromVarValue>0 && toRegValue>baseIndex)
```
Comment (ifellows) [inline inst/include/Stats.h:1410]: Note: These conditional statements can be simplified.

**Review by ifellows — COMMENTED** [review]
Note: If I mark something with "note:" it is just an FYI, not a strongly suggested change.

File: tests/testthat/test-stats.R (hunk 5 above)
Comment (duncan-clark) [reply tests/testthat/test-stats.R:198]: good catch - will add in

**Comment by duncan-clark** [comment]
Slight update fixing tests, and simplified logisticNeighbours

**Comment by ifellows** [comment]
go ahead and merge when ready

**Pull request merged**